### PR TITLE
A/B Test for Signup UX

### PR DESCRIPTION
**Pull request by biyeun**
Adds a new user signup experience where 50% of users are directed to a workflow that firsts asks whether they would like to start a trial or get a demo. Get a demo will trigger the Schedule a Demo workflow. Start Trial brings the user into the current signup ux.

@dimagi/product @dimagi/ux 

Ok the experience is up on staging

Desktop:
screen shot 2018-12-18 at 3 20 03 pm

Desktop after clicking "view features"
screen shot 2018-12-18 at 3 20 16 pm

Mobile
screen shot 2018-12-18 at 3 20 29 pm
screen shot 2018-12-18 at 3 20 34 pm

Mobile (after clicking "view features")
screen shot 2018-12-18 at 3 20 41 pm

### CODE REVIEW

**Comment by czue** [comment]
These look awesome!

One minor reaction I had is that I'm not sure how I feel about a hover-that-changes-the-text paradigm. Is that a common practice? 

It's definitely fun/cool, but for me, it's not very discoverable - i.e. I wouldn't know I could/should hover over that text and expect it to change just by looking at it. Also I worry it could lead to bad UX, e.g. if I wanted to highlight or copy/paste the text and it changed out from under me as I went to click it. 

My suggestion would be that we either keep those visible by default on larger screens or hide them under a clickable target, e.g. something like this from Box's [mobile pricing page](https://www.box.com/pricing)

image 

(after clicking "see features"): 

image

Note that we could also then apply that same paradigm to mobile if we wanted to save ourselves some vertical space.

Also it took me a while to realize that lion's head photo was your desktop and not some new design... :man_facepalming:

**Comment by biyeun** [comment]
@czue these are all valid frustrations and i'll go over them with the growth team when we all sync today.

**Comment by biyeun** [comment]
@czue regardless...this is an a/b test so the changes are not yet permanent. also, we're not testing the discoverability of the hover over features (which mimic the dimagi.com pricing page experience), but the demo vs trial split

**Review by gbova — APPROVED** [review]
I don't know much about `.less` files (those don't make too much sense to me) or UX, so I can't way in about the hover-over features. But this looks good to me! The screenshots are awesome!

File: corehq/apps/hqwebapp/static/hqwebapp/js/base_main.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ hqDefine("hqwebapp/js/base_main", [
     'analytix/js/hubspot',
     'analytix/js/kissmetrix',
     'hqwebapp/js/mobile_experience_warning',
+    'hqwebapp/js/modernizr',
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/static/hqwebapp/js/base_main.js]: Why is this being added to base_main.js? This is only for scripts that are ~always included in base.html; I thought this PR only affected the signup flow.

File: corehq/apps/hqwebapp/static/hqwebapp/js/modernizr.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+/* eslint-disable */
+hqDefine("hqwebapp/js/modernizr", [], function () {
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/static/hqwebapp/js/modernizr.js]: Can you get this from bower instead?

File: corehq/apps/hqwebapp/static/registration/less/registration-main.less [hunk 3]
```diff
@@ -187,3 +187,246 @@ p.validation-message-block {
...
+
+  h2 {
+    text-align: center;
+    color: white;
+    margin-bottom: 20px;
+  }
+}
+
+.reg-start-container .check-list {
+  @_text-indent: 8px;
```
Comment (orangejenny) [inline corehq/apps/hqwebapp/static/registration/less/registration-main.less]: Stylistic, but the `@_` is kind of funny-looking. Why not just `@text-indent`? Most of our LESS variables are "private", which I think is fairly obvious when you see them defined inside of a block.

File: corehq/apps/hqwebapp/static/hqwebapp/js/base_main.js (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/hqwebapp/static/hqwebapp/js/base_main.js]: Nevermind, I read further down and see what you're doing here.

File: corehq/apps/registration/static/registration/js/register_new_user.js [hunk 4]
```diff
@@ -2,6 +2,29 @@
...
     var initial_page_data = hqImport('hqwebapp/js/initial_page_data').get;
 
+    $('#js-start-trial').click(function (e) {
+        e.preventDefault();
+        $('#registration-start-container').addClass('hide');
+        $('#registration-form-container').fadeIn();
+    });
+
+    var kissmetrics = hqImport('analytix/js/kissmetrix');
+    kissmetrics.whenReadyAlways(function () {
```
Comment (orangejenny) [inline corehq/apps/registration/static/registration/js/register_new_user.js]: Why does this need `whenReadyAlways`? `track` is already promise-based: https://github.com/dimagi/commcare-hq/blob/33ac8470f632053f5b69a3a38b3af4521e5090fb/corehq/apps/analytics/static/analytix/js/kissmetrix.js#L124

File: corehq/apps/registration/templates/registration/register_new_user.html [hunk 5]
```diff
@@ -134,8 +145,9 @@ <h2>{% trans 'Loading...just a second!' %}</h2>
 
             </div>
         {% endif %}
-      </div>
-    </div>
-  </div>
-</div>
+      </div> {# registration-form-container #}
+
+    </div> {# .col-xs-12 #}
```
Comment (orangejenny) [inline corehq/apps/registration/templates/registration/register_new_user.html]: `{# .col-xs-12 #}` => I like this pattern

**Comment by orangejenny** [comment]
About the hover behavior, my biggest concern is usability on mobile devices, and you've handled that really well.

The flip animation is delightful, and we could use more delight, but I'd be more in favor if it didn't affect access to potentially useful content. I agree with @czue that it's not ideal to disrupt users' expectations about text being highlight-able and copy-able, even if users aren't likely to want to copy this specific text.

I'd be more in favor of the animation being purely fun. Like, if just the icons themselves flipped or animated on hover. Then if you're on a hover-capable device, you get the fun, but if you're not on a hover-capable device, no big deal, less fun but you still have all the information you need. That'd also make it less important or maybe entirely unnecessary to have the Modernizr library and the multiple sets of styles - that code is well-written, but it's a non-trivial amount of code and I'm not 100% convinced its value is worth the complexity.

**Comment by orangejenny** [comment]
^ This feedback isn't blocking. I do think it's worth pulling modernizr into bower before merging, though.

**Comment by biyeun** [comment]
@orangejenny thanks! will make the changes tomorrow. also I think quite a few of us are now in alignment to scrap the flip so I will do that as well. marketing will just have to deal 😛

**Comment by czue** [comment]
I agree with @orangejenny on the delightfulness and really like her thoughts there.

Also I could do a better job being more explicit about this but it's safe to assume that all feedback on product/ux I provide is provided in hopes it is useful, but never blocking / always fine to ignore and I'll make sure to make a stink about it if I really feel strongly.

**Comment by biyeun** [comment]
@orangejenny @czue updated to remove flip cards!

**Comment by biyeun** [comment]
@czue I think your thoughts were super useful, too, and I think we should likely make these same changes on dimagi.com's pricing page.

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
kicked builds

**Pull request merged**